File: src/pokerbots/spam_framework/Deck.java
package pokerbots.spam_framework;

import java.util.ArrayList;

public class Deck {

	ArrayList<String> cardIds;
	
	public Deck() {
		cardIds = new ArrayList<String>(Card.getCardSet());
	}
	
	public Card draw() {
		if (cardIds.size() > 0) {
			int rand =  0 + (int)(Math.random() * ((cardIds.size()-1 - 0) + 1));
			return new Card(cardIds.remove(rand));
		} else {
			return null;
		}
	}
}

File: spambot/SPAMFramework/Deck.cpp
#include "Deck.h"

Deck::Deck() {
	std::vector<std::string> list(Card::GetCardList());
	m_card_ids = list;
}

Card* Deck::Draw() {
	if(m_card_ids.size() > 0) {
		int random = 0 + (rand() % (int)(m_card_ids.size() - 0 + 1));
		Card *c = new Card(m_card_ids.erase(m_card_ids.begin() + random));
		return c;
	} else {
		return NULL;
	}
}
